### ISSUE TITLE
Text file busy

### ISSUE THREAD

**Issue by prettynatty** [os72/protoc-jar#8]
Sometimes (~50%) builds on our build server are failed because of ther following error:

```
[05:21:29][Step 1/1] Caused by: java.io.IOException: Cannot run program "/tmp/protoc7238484964672343287.exe": error=26, Text file busy
[05:21:29][Step 1/1]    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
[05:21:29][Step 1/1]    at com.github.os72.protocjar.Protoc.runProtoc(Protoc.java:69)
[05:21:29][Step 1/1]    at com.github.os72.protocjar.Protoc.runProtoc(Protoc.java:53)
[05:21:29][Step 1/1]    at im.actor.Build$$anonfun$defaultSettingsServer$5$$anonfun$apply$2.apply(Build.scala:74)
[05:21:29][Step 1/1]    at im.actor.Build$$anonfun$defaultSettingsServer$5$$anonfun$apply$2.apply(Build.scala:73)
[05:21:29][Step 1/1]    at com.trueaccord.scalapb.compiler.PosixProtocDriver.buildRunner(Process.scala:69)
[05:21:29][Step 1/1]    at com.trueaccord.scalapb.ScalaPbPlugin$$anonfun$10$$anonfun$apply$2.apply(ScalaPbPlugin.scala:54)
[05:21:29][Step 1/1]    at com.trueaccord.scalapb.ScalaPbPlugin$$anonfun$10$$anonfun$apply$2.apply(ScalaPbPlugin.scala:54)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$.executeProtoc(ProtobufPlugin.scala:68)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$.sbtprotobuf$ProtobufPlugin$$compile(ProtobufPlugin.scala:86)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$$anonfun$sourceGeneratorTask$1$$anonfun$5.apply(ProtobufPlugin.scala:115)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$$anonfun$sourceGeneratorTask$1$$anonfun$5.apply(ProtobufPlugin.scala:114)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$1.apply(Tracked.scala:235)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$1.apply(Tracked.scala:235)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3$$anonfun$apply$4.apply(Tracked.scala:249)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3$$anonfun$apply$4.apply(Tracked.scala:245)
[05:21:29][Step 1/1]    at sbt.Difference.apply(Tracked.scala:224)
[05:21:29][Step 1/1]    at sbt.Difference.apply(Tracked.scala:206)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3.apply(Tracked.scala:245)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3.apply(Tracked.scala:244)
[05:21:29][Step 1/1]    at sbt.Difference.apply(Tracked.scala:224)
[05:21:29][Step 1/1]    at sbt.Difference.apply(Tracked.scala:200)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2.apply(Tracked.scala:244)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2.apply(Tracked.scala:242)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$$anonfun$sourceGeneratorTask$1.apply(ProtobufPlugin.scala:117)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$$anonfun$sourceGeneratorTask$1.apply(ProtobufPlugin.scala:112)
[05:21:29][Step 1/1]    at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:35)
[05:21:29][Step 1/1]    at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:34)
[05:21:29][Step 1/1]    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
[05:21:29][Step 1/1]    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
[05:21:29][Step 1/1]    at sbt.std.Transform$$anon$4.work(System.scala:63)
[05:21:29][Step 1/1]    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[05:21:29][Step 1/1]    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[05:21:29][Step 1/1]    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[05:21:29][Step 1/1]    at sbt.Execute.work(Execute.scala:235)
[05:21:29][Step 1/1]    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[05:21:29][Step 1/1]    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[05:21:29][Step 1/1]    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[05:21:29][Step 1/1]    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
[05:21:29][Step 1/1]    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[05:21:29][Step 1/1]    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[05:21:29][Step 1/1]    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[05:21:29][Step 1/1]    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[05:21:29][Step 1/1]    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[05:21:29][Step 1/1]    at java.lang.Thread.run(Thread.java:745)
[05:21:29][Step 1/1] Caused by: java.io.IOException: error=26, Text file busy
[05:21:29][Step 1/1]    at java.lang.UNIXProcess.forkAndExec(Native Method)
[05:21:29][Step 1/1]    at java.lang.UNIXProcess.<init>(UNIXProcess.java:248)
[05:21:29][Step 1/1]    at java.lang.ProcessImpl.start(ProcessImpl.java:134)
[05:21:29][Step 1/1]    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
[05:21:29][Step 1/1]    at com.github.os72.protocjar.Protoc.runProtoc(Protoc.java:69)
[05:21:29][Step 1/1]    at com.github.os72.protocjar.Protoc.runProtoc(Protoc.java:53)
[05:21:29][Step 1/1]    at im.actor.Build$$anonfun$defaultSettingsServer$5$$anonfun$apply$2.apply(Build.scala:74)
[05:21:29][Step 1/1]    at im.actor.Build$$anonfun$defaultSettingsServer$5$$anonfun$apply$2.apply(Build.scala:73)
[05:21:29][Step 1/1]    at com.trueaccord.scalapb.compiler.PosixProtocDriver.buildRunner(Process.scala:69)
[05:21:29][Step 1/1]    at com.trueaccord.scalapb.ScalaPbPlugin$$anonfun$10$$anonfun$apply$2.apply(ScalaPbPlugin.scala:54)
[05:21:29][Step 1/1]    at com.trueaccord.scalapb.ScalaPbPlugin$$anonfun$10$$anonfun$apply$2.apply(ScalaPbPlugin.scala:54)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$.executeProtoc(ProtobufPlugin.scala:68)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$.sbtprotobuf$ProtobufPlugin$$compile(ProtobufPlugin.scala:86)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$$anonfun$sourceGeneratorTask$1$$anonfun$5.apply(ProtobufPlugin.scala:115)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$$anonfun$sourceGeneratorTask$1$$anonfun$5.apply(ProtobufPlugin.scala:114)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$1.apply(Tracked.scala:235)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$1.apply(Tracked.scala:235)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3$$anonfun$apply$4.apply(Tracked.scala:249)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3$$anonfun$apply$4.apply(Tracked.scala:245)
[05:21:29][Step 1/1]    at sbt.Difference.apply(Tracked.scala:224)
[05:21:29][Step 1/1]    at sbt.Difference.apply(Tracked.scala:206)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3.apply(Tracked.scala:245)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3.apply(Tracked.scala:244)
[05:21:29][Step 1/1]    at sbt.Difference.apply(Tracked.scala:224)
[05:21:29][Step 1/1]    at sbt.Difference.apply(Tracked.scala:200)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2.apply(Tracked.scala:244)
[05:21:29][Step 1/1]    at sbt.FileFunction$$anonfun$cached$2.apply(Tracked.scala:242)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$$anonfun$sourceGeneratorTask$1.apply(ProtobufPlugin.scala:117)
[05:21:29][Step 1/1]    at sbtprotobuf.ProtobufPlugin$$anonfun$sourceGeneratorTask$1.apply(ProtobufPlugin.scala:112)
[05:21:29][Step 1/1]    at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:35)
[05:21:29][Step 1/1]    at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:34)
[05:21:29][Step 1/1]    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
[05:21:29][Step 1/1]    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
[05:21:29][Step 1/1]    at sbt.std.Transform$$anon$4.work(System.scala:63)
[05:21:29][Step 1/1]    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[05:21:29][Step 1/1]    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[05:21:29][Step 1/1]    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[05:21:29][Step 1/1]    at sbt.Execute.work(Execute.scala:235)
[05:21:29][Step 1/1]    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[05:21:29][Step 1/1]    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[05:21:29][Step 1/1]    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[05:21:29][Step 1/1]    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
[05:21:29][Step 1/1]    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[05:21:29][Step 1/1]    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[05:21:29][Step 1/1]    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[05:21:29][Step 1/1]    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[05:21:29][Step 1/1]    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[05:21:29][Step 1/1]    at java.lang.Thread.run(Thread.java:745)
```

build server is running on Ubuntu 14.04. protoc-jar 3.0.0-b2

**Comment by os72** [comment]
Looking.. It could be because I'm not closing the stream in `extractProtoc`. If I make a fix, could you make local SNAPSHOT build and help verify?

**Comment by prettynatty** [comment]
yep

**Comment by os72** [comment]
Actually the streams are closed properly in `finally` block
Are you running some sort of parallel builds? Could you modify the setting and try?

**Comment by os72** [comment]
It appears to be a JDK bug on Linux, not sure how to work around
https://bugs.openjdk.java.net/browse/JDK-8068370

**Comment by os72** [comment]
You could try out the change on branch issue_text_file_busy
https://github.com/os72/protoc-jar/commits/issue_text_file_busy

Please feel free to research and experiment further if the change doesn't fix it

**Comment by prettynatty** [comment]
Well, it seems that problem has gone! None of ~20 latest builds failed with protoc-jar published from this branch.

**Comment by os72** [comment]
That's great! Is it consistent for you: previous version still shows issue, new version works fine? (Would be good to verify it's because of the code change, not some change in your environment)

**Comment by prettynatty** [comment]
Yes. We didn' change anything in our CI environment, just switched to protoc-jar compiled from `issue_text_file_busy` branch.

**Comment by os72** [comment]
Okay great, I'll make an official release then

**Comment by os72** [comment]
Released protoc-jar-3.0.0-b2.1

**Comment by prettynatty** [comment]
Thanks!

**Comment by os72** [comment]
Issue resurfaced in protoc-jar-maven-plugin
https://github.com/os72/protoc-jar-maven-plugin/issues/33

**Comment by os72** [comment]
Released 3.2.0
